main.go

```diff
@@ -42,19 +42,27 @@ func main() {
 		log.Fatal(err)
 	}
 
-	http.HandleFunc("/token", tokenRequestHandler)
+	http.Handle("/token", handler{tokenRequestHandler})
 	go func() {
-		http.ListenAndServe(":80", nil)
+		log.Fatal(http.ListenAndServe(":80", nil))
 	}()
 
 	log.Println("Listening for token requests.")
-	signalChan := make(chan os.Signal, 1)
-	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
-	for {
-		select {
-		case <-signalChan:
-			log.Printf("Shutdown signal received, exiting...")
-			os.Exit(0)
-		}
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
+	<-quit
+	log.Printf("Shutdown signal received, exiting...")
+}
+
+type handler struct {
+	f func(io.Writer, *http.Request) (int, error)
+}
```

Comment:
I know this pattern, but I feel like implementing methods on functions can be confusing for non gophers, plus it doesn't compose as well as structs.